specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+  /// Remove an event handler previously added with add_ClientCertificateRequested.
+  HRESULT remove_ClientCertificateRequested([in] EventRegistrationToken token);
+}
+
+/// Provides access to the certificate metadata
+[uuid(9ca02116-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2ClientCertificate : IUnknown {
+  /// Subject of the certificate.
+  [propget] HRESULT Subject([out, retval] LPWSTR* subject);
+  /// Name of the certificate authority that issued the certificate.
+  [propget] HRESULT Issuer([ out, retval ] LPWSTR* issuer);
+  /// The valid start date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidStartDate([out, retval] double* validStartDate);
+  /// The valid expiration date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidExpiryDate([out, retval] double* validExpiryDate);
+  /// DER encoded serial number of the certificate.
+  [propget] HRESULT SerialNumber([out, retval] LPWSTR* serialNumber);
+  /// Display name for a certificate.
+  [propget] HRESULT DisplayName([out, retval] LPWSTR* displayName);
+  /// PEM encoded data for the certificate.
+  /// Returns Base64 encoding of DER encoded certificate.
+  /// Read more about PEM at [RFC 1421 Privacy Enhanced Mail]
+  /// (https://tools.ietf.org/html/rfc1421).
+  [propget] HRESULT PEMEncodedData([out, retval] LPWSTR* pemEncodedData);
+  /// List of PEM encoded client certificate issuer chain.
+  /// This list contains the certificate and intermediate CA certificates.
+  [propget] HRESULT PEMEncodedIssuerChainData([out, retval]
+      ICoreWebView2ClientCertificateIssuerChainList**
+                                              pemEncodedIssuerChainData);
+  /// Kind of a certificate (smart card, pin).
+  [propget] HRESULT CertificateKind([out,retval]
+      COREWEBVIEW2_CLIENT_CERTIFICATE_KIND* certificateKind);
+}
+
+/// A list of distinguished certificate authorities.
+[uuid(c87bb63e-a3f4-11eb-bcbc-0242ac130002), object, pointer_default(unique)]
+interface ICoreWebView2CertificateAuthorityList : IUnknown {
+  /// The number of certificate authorities contained in
+  /// ICoreWebView2CertificateAuthorityList.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the certificate authority at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] LPWSTR* certificateAuthority);
+}
+
+/// A list of client certificate issuer chains.
+[uuid(4271275e-9107-11eb-a8b3-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2ClientCertificateIssuerChainList : IUnknown {
+  /// The number of issuer chains contained in
+  /// ICoreWebView2ClientCertificate.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets encoded data of the issuer chain at the given index.
+  /// Index 0 contains the certificate followed by intermediate CA certificates.
```

Comment:
Not sure what this is saying. Could mean

* At index 0, you get a certificate concatenated with intermediate CA certificates. (No information about what certificates are at indices 1 through Count&minus;1.)
* At index 0 is a certificate, and at indices 1 through Count&minus;1 are the intermediate CA certificates. (No information about what kind of certificate is at index 0.)

Reply:
I think this is saying index 0 contains this certificate followed by intermediate certificates in order with the root certificate at position Count - 1.

Either way, please update the comment to make it clearer.